service/storage_service.cc

```diff
@@ -6846,7 +6846,7 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
             throw std::runtime_error(format("Unknown host: {}", dst.host));
         }
         if (dst.shard >= node->get_shard_count()) {
-            throw std::runtime_error(format("Host {} does not have shard {}", dst.shard));
+            throw std::runtime_error(format("Host {} does not have shard {}", *node, dst.shard));
```

Comment:
what's left to enable compile-time checking?

Reply:
@avi see https://github.com/scylladb/seastar/pull/1196